### ISSUE TITLE
async arrow functions use wrong `arguments` to construct rest params

### ISSUE THREAD

**Issue by jbt** [babel/babel#1800]
```js
let fn = async (...rest) => rest;
```

compiles to:

```js
"use strict";

var _arguments = arguments,
    _this = this;

var fn = function fn() {
  var _len,
      rest,
      _key,
      args$1$0 = _arguments;

  return regeneratorRuntime.async(function fn$(context$1$0) {
    while (1) switch (context$1$0.prev = context$1$0.next) {
      case 0:
        for (_len = args$1$0.length, rest = Array(_len), _key = 0; _key < _len; _key++) {
          rest[_key] = args$1$0[_key];
        }

        return context$1$0.abrupt("return", rest);

      case 2:
      case "end":
        return context$1$0.stop();
    }
  }, null, _this);
};
```

For some reason `args$1$0` is being constructed from `_arguments` pulled from the outer function's arguments rather than the args to `fn`. (well, in this reduced case it'll actually ReferenceError on undefined `arguments`, but the case I originally discovered was inside another function, so it was doing Weird Stuff)

This seems to have changed fairly recently, it doesn't show up on the [repl](https://babeljs.io/repl/#?experimental=true&evaluate=false&loose=true&spec=false&playground=true&code=let%20fn%20%3D%20async%20(...rest)%20%3D%3E%20rest%3B) which is still using 5.5.1. Likewise everything up to and including 5.5.8 from npm behaves fine.

**Comment by jbt** [comment]
Holy crap that was quick. Awesome! :+1:

**Comment by sebmck** [comment]
Fixed as of 5.6.3
